src/Mod/Fem/Gui/ViewProviderFemConstraint.cpp

```diff
@@ -115,6 +123,32 @@ void ViewProviderFemConstraint::attach(App::DocumentObject* pcObject)
     addDisplayMaskMode(sep, "Base");
 }
 
+std::string ViewProviderFemConstraint::resourceSymbolDir =
+    App::Application::getResourceDir() + "Mod/Fem/Resources/symbols/";
+
+void ViewProviderFemConstraint::loadSymbol(const char* fileName)
+{
+    ivFile = fileName;
+    SoInput* in = new SoInput;
```

Comment:
There is no need to create the SoInput on the heap. Just create it on the stack and pass it as `&in` to readAll().

Reply:
Done